nixos/lib/systemd-types.nix

```diff
@@ -40,7 +40,7 @@ rec {
       enable = mkEnableOption (lib.mdDoc "copying of this file and symlinking it") // { default = true; };
 
       target = mkOption {
-        type = types.path;
+        type = types.str;
```

Comment:
If I remember correctly `types.path` also accepts strings that represent paths. So this generalizes what is already accepted.

Reply:
@mweinelt the issue with `types.path` was that it doesn't work for `boot.initrd.environment.etc`, since it requires the first character to be `/`. Though, as I said before, I'm thinking we can probably get rid of `boot.initrd.environment.etc` because I bared even used it in this PR and it's easy to deal without.